pkg/kubelet/cm/devicemanager/device_plugin_stub.go

```diff
@@ -104,6 +110,43 @@ func (m *Stub) Stop() error {
 	return m.cleanup()
 }
 
+// GetInfo is the RPC which return pluginInfo
+func (m *Stub) GetInfo(stream watcherapi.Registration_GetInfoServer) error {
+	log.Println("GetInfo")
+	info := watcherapi.PluginInfo{
+		Type:    watcherapi.DevicePlugin,
+		Name:    m.resourceName,
+		Version: []string{pluginapi.Version},
+	}
+	err := stream.Send(&info)
+	if err != nil {
+		log.Println("error sending plugin info: ", err)
+	}
+
+	for {
+		s, err := stream.Recv()
+		log.Println("Got ", s)
+		if err == io.EOF {
+			log.Println("Breaking err ", err)
+			break
+		}
+		if err != nil {
+			log.Println("Received err ", err)
+			return err
+		}
+		if s.PluginRegistered {
+			log.Println("Registration success!!")
+			break
+		}
+		if s.Error != "" {
+			log.Println(s.Error)
+			break
+		}
+	}
+
+	return nil
+}
+
 // Register registers the device plugin for the given resourceName with Kubelet.
 func (m *Stub) Register(kubeletEndpoint, resourceName string) error {
```

Comment:
Remove this function?

Reply:
Changes are backward compatible, so this function is still required.